### ISSUE TITLE
4.4.0 release (getch.c): compile warnings about deprecated timeb breaking vt compile

### ISSUE THREAD

**Issue by GitMensch** [Bill-Gray/PDCursesMod#311]
The warnings also are raised with other environments, but as those don't use `-Werror` we do not have an error during make.

```
gcc -Wall -Wextra -pedantic -Werror -g -DPDCDEBUG -fPIC -DPDC_DLL_BUILD -I.. -DHAVE_VSNPRINTF -DHAVE_VSSCANF -c ../pdcurses/getch.c
../pdcurses/getch.c: In function 'PDC_millisecs':
../pdcurses/getch.c:419:12: error: 'timeb' is deprecated [-Werror=deprecated-declarations]
  419 |     struct timeb t;
      |            ^~~~~
In file included from ../pdcurses/getch.c:415:
c:\mingw\include\sys\timeb.h:57:32: note: declared here
   57 | struct __POSIX_2001_DEPRECATED timeb
      |                                ^~~~~
../pdcurses/getch.c:421:5: error: 'ftime' is deprecated [-Werror=deprecated-declarations]
  421 |     ftime( &t);
      |     ^~~~~
In file included from ../pdcurses/getch.c:415:
c:\mingw\include\sys\timeb.h:169:6: note: declared here
  169 | void ftime (struct timeb *__t) {_ftime ((struct _timeb *)(__t)); }
      |      ^~~~~
cc1.exe: all warnings being treated as errors
make: *** [getch.o] Error 1
```

The reason is that MinGW was included in the deprecated ftime path with 6b432b4dc40350fc9576bb269c0c66bdad751289.

The solution seems to be (not verified yet) adding a

```c
#ifdef _POSIX_C_SOURCE >= 199309L
clock_gettime( CLOCK_REALTIME, &t);
#elif ... //  current code here
...
```

**Comment by Bill-Gray** [comment]
You are correct.  A check is required to make sure that `_POSIX_C_SOURCE` is defined.  However,  it does make more sense,  overall,  to apply the feature macros that test for the availability of `clock_gettime()` and/or `gettimeofday()` than it does to put together some rather ad hoc tests to say we should use `ftime()`.

I have been testing [this modified `getch.c`](https://www.projectpluto.com/temp/getch.c),  which uses `clock_gettime()` if available,  `gettimeofday()` otherwise,  and only dropping to `ftime()` if neither of those are available.  Still tryng it out,  but expect to be able to commit the change,  push it,  and close this issue.

**Comment by GitMensch** [comment]
Sounds good - but I highly suggest to use a standard name instead (and wrap it into one conditional):

```c
#if  defined( _DEFAULT_SOURCE) || defined( _BSD_SOURCE) || \
   ( defined( _POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 199309L )
   #define GETTIMEOFDAY_AVAILABLE
#endif
```

**Comment by GitMensch** [comment]
Sounds good - but I highly suggest to use a standard name instead (and wrap it into one conditional):

```c
#if  defined (HAVE_CLOCK_GETTIME) || \
   (defined( _POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 199309L)

 #include <time.h>

long PDC_millisecs( void)
{
    struct timespec t;

    clock_gettime( CLOCK_REALTIME, &t);
    return( t.tv_sec * 1000 + t.tv_nsec / 1000000);
}

#elif defined (HAVE_GETTIMEOFDAY) || \\
  defined( _DEFAULT_SOURCE) || defined( _BSD_SOURCE)

 #include <sys/time.h>

long PDC_millisecs( void)
{
    struct timeval t;

    gettimeofday( &t, NULL);
    return( t.tv_sec * 1000 + t.tv_usec / 1000);
}

#else    /* neither gettimeofday() or clock_gettime() available */

 #include <sys/timeb.h>

long PDC_millisecs( void)
{
    struct timeb t;

    ftime( &t);
    return( (long)t.time * 1000L + (long)t.millitm);
}
#endif
```

**Comment by GitMensch** [comment]
Fixed by 45949000c3ac1375f5f821d72f46e4726a3a6a2f - just not released.
